### ISSUE TITLE
utils/hash - process_file() bug

### ISSUE THREAD

**Issue by kasicass** [OurMachinery/themachinery-public#358]
## bug

I play with hash.exe, and found that "TM_STATIC_HASH("baby", 2)" doesn't been converted.

```C
/* hello.c */
#include <stdio.h>

TM_STATIC_HASH("hello", 1)
TM_STATIC_HASH("baby", 2)
```

## fix

if buffer_len != macro_len, it seams "o += buffer_len" is meaningless.

```C
        const uint64_t hash = tm_murmur_hash_64a_inline(string_s, string_len, 0);
        sprintf(buffer, "TM_STATIC_HASH(\"%.*s\", 0x%" PRIx64 "ULL)", string_len, string_s, hash);
        const uint32_t buffer_len = (uint32_t)strlen(buffer);
        if (o + buffer_len > size || 0 == memcmp(macro_s, buffer, buffer_len)) {
            o += buffer_len;
            continue;
        }
```

=>

```C
        const uint64_t hash = tm_murmur_hash_64a_inline(string_s, string_len, 0);
        sprintf(buffer, "TM_STATIC_HASH(\"%.*s\", 0x%" PRIx64 "ULL)", string_len, string_s, hash);
        const uint32_t buffer_len = (uint32_t)strlen(buffer);
        if (buffer_len == macro_len && 0 == memcmp(macro_s, buffer, buffer_len)) {
            o += buffer_len;
            continue;
        }
```

**Comment by niklas-ourmachinery** [comment]
Hmm... `TM_STATIC_HASH("baby", 2)` seems to work for me... which version of the engine is this?

**Comment by kasicass** [comment]
I use beta-2020.11

```
D:\3dengine\beta-2020.11\bin>type hello.c
#include <stdio.h>

TM_STATIC_HASH("hello", 1)
TM_STATIC_HASH("baby", 2)

D:\3dengine\beta-2020.11\bin>hash hello.c
hello.c:3 TM_STATIC_HASH("hello", 0x1e68d17c457bf117ULL)

D:\3dengine\beta-2020.11\bin>type hello.c
#include <stdio.h>

TM_STATIC_HASH("hello", 0x1e68d17c457bf117ULL)
TM_STATIC_HASH("baby", 2)

D:\3dengine\beta-2020.11\bin>
```

**Comment by niklas-ourmachinery** [comment]
Thanks, will have a look at it.

**Comment by niklas-ourmachinery** [comment]
Fixed for next release (2020.11.a), thanks for the diff!
